ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/avdipaddressing.py

```diff
@@ -176,8 +182,14 @@ def p2p_uplinks_peer_ip(self, uplink_switch_index: int) -> str:
             )
 
         prefixlen = self._fabric_ip_addressing_p2p_uplinks_ipv4_prefix_length
-        offset = ((self._id - 1) * self._max_uplink_switches * self._max_parallel_uplinks) + uplink_switch_index
-        return get_ip_from_pool(self._uplink_ipv4_pool, prefixlen, offset, 0)
+        p2p_ipv4_pool = self._get_p2p_ipv4_pool(uplink_switch_index)
+
+        if self.shared_utils.uplink_ipv4_pool is None:
+            # Using downlink_ipv4_pool from uplink_switch
+            offset = ((self._id - 1) * self._max_parallel_uplinks) + self._get_parallel_uplink_index(uplink_switch_index)
```

Comment:
As mentioned elsewhere, this should be using the index of our uplink_switch_interface in the ranges associated with the downlink_pool